lib/liquid/expression.rb

```diff
@@ -21,20 +21,24 @@ def to_liquid
       'empty'.freeze => MethodLiteral.new(:empty?, '').freeze
     }
 
+    SINGLE_QUOTED_STRING = /\A'(.*)'\z/m
+    DOUBLE_QUOTED_STRING = /\A"(.*)"\z/m
+    INTEGERS_REGEX       = /\A(-?\d+)\z/
+    FLOATS_REGEX         = /\A(-?\d+\.\d+)\z/
```

Comment:
So, I know this was weird before, allowing `1.2.3` as a float, but I'm not sure it's 100% safe for us to change it without some testing against our production templates. Can we keep it weird for now?

Liquid-C's strict parsing [will raise an error](https://github.com/Shopify/liquid-c/blob/958052d08fb857f16a6799dbe5edeea092b54a67/ext/liquid_c/parser.c#L131-L158) in that case FWIW, though in Shopify we swallow that error and run the lax parser.

Reply:
I've reverted back to the "weird" regex.
Funny how it also matches `1..10` as a `Float`.. :slightly_smiling_face: